src/display/editor/highlight.js

```diff
@@ -0,0 +1,491 @@
+      ],
+      [
+        AnnotationEditorParamsType.HIGHLIGHT_OPACITY,
+        Math.round(HighlightEditor._defaultOpacity * 100),
+      ],
+    ];
+  }
+
+  /** @inheritdoc */
+  get propertiesToUpdate() {
+    return [
+      [
+        AnnotationEditorParamsType.HIGHLIGHT_COLOR,
+        this.#color || HighlightEditor._defaultColor,
+      ],
+      [
+        AnnotationEditorParamsType.HIGHLIGHT_OPACITY,
+        Math.round(100 * (this.#opacity ?? HighlightEditor._defaultOpacity)),
+      ],
+    ];
+  }
+
+  /**
+   * Update the color and make this action undoable.
+   * @param {string} color
+   */
+  #updateColor(color) {
+    const savedColor = this.color;
+    this.addCommands({
+      cmd: () => {
+        this.#color = color;
+        this.parent.drawLayer.changeColor(this.#id, color);
+      },
+      undo: () => {
+        this.#color = savedColor;
+        this.parent.drawLayer.changeColor(this.#id, savedColor);
+      },
+      mustExec: true,
+      type: AnnotationEditorParamsType.HIGHLIGHT_COLOR,
+      overwriteIfSameType: true,
+      keepUndo: true,
+    });
+  }
+
+  /**
+   * Update the opacity and make this action undoable.
+   * @param {number} opacity
+   */
+  #updateOpacity(opacity) {
+    opacity /= 100;
+    const savedOpacity = this.#opacity;
+    this.addCommands({
+      cmd: () => {
+        this.#opacity = opacity;
+        this.parent.drawLayer.changeOpacity(this.#id, opacity);
+      },
+      undo: () => {
+        this.#opacity = savedOpacity;
+        this.parent.drawLayer.changeOpacity(this.#id, savedOpacity);
+      },
+      mustExec: true,
+      type: AnnotationEditorParamsType.HIGHLIGHT_OPACITY,
+      overwriteIfSameType: true,
+      keepUndo: true,
+    });
+  }
+
+  /** @inheritdoc */
+  fixAndSetPosition() {
+    const [pageWidth, pageHeight] = this.pageDimensions;
+    let { x, y, width, height } = this;
+    width *= pageWidth;
+    height *= pageHeight;
+    x *= pageWidth;
+    y *= pageHeight;
+
+    x = Math.max(0, Math.min(pageWidth - width, x));
+    y = Math.max(0, Math.min(pageHeight - height, y));
+
+    this.x = x /= pageWidth;
+    this.y = y /= pageHeight;
+
+    const { style } = this.div;
+    style.left = `${(100 * x).toFixed(2)}%`;
+    style.top = `${(100 * y).toFixed(2)}%`;
+
+    this.moveInDOM();
+  }
+
+  /** @inheritdoc */
+  getRect(tx, ty) {
+    const scale = this.parentScale;
+    const [pageWidth, pageHeight] = this.pageDimensions;
+    const [pageX, pageY] = this.pageTranslation;
+    const shiftX = tx / scale;
+    const shiftY = ty / scale;
+    const x = this.x * pageWidth;
+    const y = this.y * pageHeight;
+    const width = this.width * pageWidth;
+    const height = this.height * pageHeight;
+
+    return [
+      x + shiftX + pageX,
+      pageHeight - y - shiftY - height + pageY,
+      x + shiftX + width + pageX,
+      pageHeight - y - shiftY + pageY,
+    ];
+  }
+
+  /** @inheritdoc */
+  isEmpty() {
+    return false;
+  }
```

Comment:
This seems to be the default value, looking at https://github.com/mozilla/pdf.js/blob/b7e943bc7d31ef4e8f19c60a02eb9e70c2d17eb7/src/display/editor/editor.js#L1278-L1284, so do we actually need this or should this one perhaps be updated?